runtime/parachains/src/shared.rs

```diff
@@ -94,16 +94,15 @@ impl<Hash: PartialEq + Copy, BlockNumber: AtLeast32BitUnsigned + Copy>
 		prev: Option<BlockNumber>,
 	) -> Option<(Hash, BlockNumber)> {
 		let pos = self.buffer.iter().position(|(rp, _)| rp == &relay_parent)?;
+		let age = (self.buffer.len() - 1) - pos;
+		let number = self.latest_number - BlockNumber::from(age as u32);
 
 		if let Some(prev) = prev {
-			if prev >= self.latest_number {
+			if prev >= number {
```

Comment:
```suggestion
			if prev > number {
```

Same relay parent is allowed, moving backwards isn't.